### PR TITLE
Sb v1.43.0

### PR DESCRIPTION
**Pull request by ScottBailey**
Update grpc-cpp with conda-forge and  v1.43.0. (From 1.39.0)

Fixed tests, removed unused patch, set compiler versions for some linux architectures, set static builds ONLY for win64 and skipped win32.

s390x does not currently build due to missing dependencies.

**Upstream:**
- https://github.com/grpc/grpc/tree/v1.43.0

**Version Info:**
- https://github.com/grpc/grpc/blob/v1.43.0/CMakeLists.txt#L28
-  https://github.com/grpc/grpc/blob/b0f37a22bbae12a4b225a88be6d18d5e41dce881/Makefile#L458

**Changelog:**
- https://github.com/grpc/grpc/releases

Reviewing the changelog, there is an issue I consider worth watching: https://github.com/grpc/grpc/issues/28513 indicates a memory leak; however, it doesn't indicate the leak is isolated to a particular version of grpc-cpp, so I am inclined to take 1.43.0.

**Testing:**

```
conda build ../wip/grpc-cpp-feedstock --error-overlinking  --test
```

results in:

```
All tests passed
```

**conda-forge issues:**
- https://github.com/conda-forge/grpc-cpp-feedstock/issues

None that affect us.

**License:**

- https://github.com/grpc/grpc/blob/master/LICENSE

```
 license: Apache-2.0
```

**dev url:**

```
  dev_url: https://github.com/grpc/grpc
```

**doc url:**

```
  doc_url: https://grpc.io/docs/
```

### CODE REVIEW

**Comment by skupr-anaconda** [comment]
An error on `390x`:
```
conda.exceptions.ResolvePackageNotFound:
  - abseil-cpp
  - re2

```
We can add a comment about these missing packages above the updated line `skip: True  # [win32 or s390x] `

**Comment by ScottBailey** [comment]
> We can add a comment about these missing packages above the updated line `skip: True # [win32 or s390x] `

Added `skip` and comment for s390x.

**Review by skupr-anaconda — APPROVED** [review]

File: recipe/build.sh [hunk 1]
```diff
@@ -2,21 +2,22 @@
 
 set -ex
 
-if [[ "$CONDA_BUILD_CROSS_COMPILATION" == 1 ]]; then
+if [[ "${target_platform}" == osx* ]]; then
+  export CMAKE_ARGS="${CMAKE_ARGS} -DCMAKE_CXX_STANDARD=14"
+else
+  export CMAKE_ARGS="${CMAKE_ARGS} -DCMAKE_CXX_STANDARD=17"
+fi
```
Comment (tobijk) [inline recipe/build.sh:14]: Is there a technical reason for this distinction? Will it not build on Linux with gcc under C++14? I see that we are explicitly selecting a more recent compiler below. Wouldn't it be better to keep a lowest common denominator across platforms?

File: recipe/conda_build_config.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+c_compiler_version:    # [linux and not aarch64]
+- '8'                  # [linux and not aarch64]
+cxx_compiler_version:  # [linux and not aarch64]
+- '8'                  # [linux and not aarch64]
```
Comment (tobijk) [inline recipe/conda_build_config.yaml]: C++17 was still marked as experimental in gcc 8, IINM. So there may be incompatible changes in libstdc++ after that.

**Review by tobijk — COMMENTED** [review]
Just a few comments, LGTM.

File: recipe/build.sh (hunk 1 above)
Comment (ScottBailey) [reply recipe/build.sh:14]: I've added documentation in the file to explain the need - it has to do with how `abseil-cpp` is built.

File: recipe/conda_build_config.yaml (hunk 2 above)
Comment (ScottBailey) [reply recipe/conda_build_config.yaml]: You are correct, C++17 remained experimental in gcc8. I have updated the compiler to gcc9 where available as well as documenting the reason for the compiler call out.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -26,17 +24,18 @@ package:
...
   # You can use module version of re2 if you add this and chage build.sh:
   # - git_url: https://github.com/google/re2
   #   git_ref: aecba11114cf1fac5497aeb844b6966106de3eb6
   #   folder: third_party/re2
 
 build:
-  number: 5
+  number: 0
+  # Anaconda's s390x does not contain depdendencies for abseil-cpp or re2, this package is skipped for s390x at this time.
+  skip: True  # [win32 or s390x]
```
Comment (tobijk) [inline recipe/meta.yaml:39]: I see we have previously supported win32, why are we skipping it now?

Comment (ScottBailey) [reply recipe/meta.yaml:39]: Latest protobuf recipe does not support 32bit.

**Review by tobijk — APPROVED** [review]
LGTM

**Pull request merged**